go/dhcp/utils.go

```diff
@@ -0,0 +1,335 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/binary"
+	"fmt"
+	"log"
+	"math/rand"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/coreos/etcd/client"
+	"github.com/inverse-inc/packetfence/go/pfconfigdriver"
+)
+
+const hexDigit = "0123456789abcdef"
+
+type NodeInfo struct {
+	Mac      string
+	Status   string
+	Category string
+}
+
+// inc function use to increment an ip
+func inc(ip net.IP) {
+	for j := len(ip) - 1; j >= 0; j-- {
+		ip[j]++
+		if ip[j] > 0 {
+			break
+		}
+	}
+}
+
+// inc function use to decrement an ip
+func dec(ip net.IP) {
+	for j := len(ip) - 1; j >= 0; j-- {
+		ip[j]--
+		if ip[j] == 255 {
+			continue
+		}
+		if ip[j] > 0 {
+			break
+		}
+	}
+}
+
+// connectDB connect to the database
+func connectDB(configDatabase pfconfigdriver.PfconfigDatabase, db *sql.DB) {
```

Comment:
This will start to be common between processes, we could have a database package for go that centralize the current DB connection logic and the future additional logic